### PR TITLE
pyproject: switch to setuptools legacy backend

### PR DESCRIPTION
**Pull request by FFY00**
I need stop being picky and just drop poerty as it doesn't make that much sense for this project. Maybe in the future...

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,50 +1,3 @@
...
-
-[tool.poetry.dev-dependencies]
-pytest = '^5.0'
-pytest-mock = '^3.0'
-pytest-cov = '^2.0'
-
-[tool.poetry.scripts]
-python-build = 'build.__main__:main'
+requires = ['setuptools >= 40.8.0', 'wheel']
+build-backend = 'setuptools.build_meta:__legacy__'
```
Comment (gaborbernat) [inline pyproject.toml:3]: Why the legacy backend?

Comment (pganssle) [reply pyproject.toml:3]: Yeah definitely don't use this. It's intended for backwards compatibility only.

Comment (FFY00) [reply pyproject.toml]: Ah, sorry. I just defaulted to `__legacy__` because it's our fallback on missing `build-backend`.

**Comment by FFY00** [comment]
Nothing made it unusable. It was another place to maintain metadata.

**Comment by gaborbernat** [comment]
I meant why not drop setuptools over poetry 😅

**Comment by FFY00** [comment]
Ah, because I want it to be easily bootstrapable.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
       run: pip install typing
 
     - name: Install build backend
-      run: pip install poetry
+      run: pip install setuptools wheel
```
Comment (gaborbernat) [inline .github/workflows/build.yml:34]: You should not need this, pip installs them on demand.

Comment (FFY00) [reply .github/workflows/build.yml:34]: Only after #18, right now we don't build in an isolated environment.

**Pull request merged**